### PR TITLE
Feat archive trash tags functionality

### PR DESCRIPTION
**Pull request by aishwarya-bm**
1. Archive - add, restore
2. Trash - add, restore
3. Tags/labels - add to notes, modify in notes, add new labels,labels tab

### CODE REVIEW

**Review by sadathshariff — APPROVED** [review]
The code looks clean, like how you are handling the user login and navigating to the login page.
Nice UI

File: src/contexts/notes-context/notes-reducer.jsx [hunk 1]
```diff
@@ -1,3 +1,6 @@
+import dayjs from "dayjs";
+
+const formatDate = () => dayjs().format("YYYY-MM-DDTHH:mm:ssZ");
```
Comment (dev-enforced) [inline src/contexts/notes-context/notes-reducer.jsx:3]: Learnt a new format of using Dates and storing it!!

File: src/contexts/notes-context/notes-reducer.jsx [hunk 2]
```diff
@@ -31,17 +41,109 @@ export default function notesReducer(state, { type, payload }) {
...
+          state.archives.find(item => item._id === payload.note_id),
+        ],
+        archives: payload.archives,
+      };
+    }
+
+    case "ADD_NEW_TAG": {
+      return {
+        ...state,
+        tags: [...state.tags, payload],
```
Comment (dev-enforced) [inline src/contexts/notes-context/notes-reducer.jsx:115]: You can have a check on the payload value whether it is already present in the tag List then no need to update the tags and if it is new value you can update the tag List. This way you can avoid adding same tag twice!!

**Review by dev-enforced — APPROVED** [review]
The code structure looks good and the preview of Slate Notes is also working: https://notes-app-git-feat-archive-trash-tags-funct-541697-aishwarya-bm.vercel.app/tags as mentioned in the PR!!
Got introduced to DayJS library and then the logic which you have used to add tags and remove tags from notes across all the 3 sections of Archives,Trash and the actual notes section
Good work @aishwarya-bm ✅

File: src/contexts/notes-context/notes-reducer.jsx (hunk 2 above)
Comment (aishwarya-bm) [reply src/contexts/notes-context/notes-reducer.jsx:115]: Thanks for pointing this out Roshan! I have handled adding duplicate tags now!

**Pull request merged**